### PR TITLE
ITx: Allow setting isolation level

### PR DESCRIPTION
**Pull request by dpetrick**
Allows setting the isolation level when starting an interactive transaction (iTx).
The configuration payload has an additional field:
```jsonc
{
    "max_wait": 1234,
    "timeout": 4321,
    "isolation_level": "Serializable" // << New
}
```

An enum is exposed on the DMMF that specifies valid values for the currently loaded connector:
```jsonc
{
  "datamodel": {
    // ...
  },
  "schema": {
    "inputObjectTypes": {
      // ...
    },
    "outputObjectTypes": {
      // ...
    },
    "enumTypes": {
      "prisma": [
        { // <<< This is added
          "name": "TransactionIsolationLevel",
          "values": [
            // <<< These are dependent on the datasource
            "ReadCommitted",
            "ReadUncommitted",
            // ...
          ]
        }
      ],
      "model": [
        // ...
      ]
    }
  },
  "mappings": {
    // ...
  }
}
```

### CODE REVIEW

File: query-engine/dmmf/src/ast_builders/schema_ast_builder/type_renderer.rs [hunk 1]
```diff
@@ -36,18 +39,19 @@ pub(super) fn render_output_type(output_type: &OutputType, ctx: &mut RenderConte
...
 
-        OutputType::Scalar(ScalarType::Enum(et)) => {
-            ctx.mark_to_be_rendered(&et.as_ref());
-            let type_reference = DmmfTypeReference {
-                typ: et.name().to_owned(),
-                namespace: Some(et.namespace()),
-                location: TypeLocation::Scalar,
-                is_list: false,
-            };
+        // OutputType::Scalar(ScalarType::Enum(et)) => {
```
Comment (garrensmith) [inline query-engine/dmmf/src/ast_builders/schema_ast_builder/type_renderer.rs:42]: Is this code still needed?

File: query-engine/dmmf/src/ast_builders/schema_ast_builder/type_renderer.rs [hunk 2]
```diff
@@ -63,7 +67,7 @@ pub(super) fn render_output_type(output_type: &OutputType, ctx: &mut RenderConte
                 ScalarType::JsonList => "Json",
                 ScalarType::Xml => "Xml",
                 ScalarType::Bytes => "Bytes",
-                ScalarType::Enum(_) => unreachable!(), // Handled separately above.
+                // ScalarType::Enum(_) => unreachable!(), // Handled separately above.
```
Comment (garrensmith) [inline query-engine/dmmf/src/ast_builders/schema_ast_builder/type_renderer.rs:70]: Can you remove?

File: query-engine/dmmf/src/ast_builders/schema_ast_builder/type_renderer.rs [hunk 3]
```diff
@@ -118,18 +125,17 @@ pub(super) fn render_input_type(input_type: &InputType, ctx: &mut RenderContext)
...
-                namespace: Some(et.namespace()),
-                location: TypeLocation::Scalar,
-                is_list: false,
-            };
-
-            type_reference
-        }
-
+        // InputType::Scalar(ScalarType::Enum(et)) => {
+        //     ctx.mark_to_be_rendered(&et.as_ref());
```
Comment (garrensmith) [inline query-engine/dmmf/src/ast_builders/schema_ast_builder/type_renderer.rs:129]: can you remove

File: query-engine/prisma-models/src/builders/internal_dm_builder.rs [hunk 4]
```diff
@@ -63,6 +64,20 @@ impl InternalDataModelBuilder {
 
         internal_data_model.relations.set(relations).unwrap();
         internal_data_model.finalize();
+
+        internal_data_model
+            .composite_types
```
Comment (garrensmith) [inline query-engine/prisma-models/src/builders/internal_dm_builder.rs:69]: How is this related to the isolation levels?

File: query-engine/core/src/response_ir/internal.rs [hunk 5]
```diff
@@ -582,13 +582,15 @@ fn convert_enum(value: PrismaValue, dbt: &DatabaseEnumType) -> Result<Item, Core
             Some(inum) => Ok(Item::Value(inum)),
             None => Err(CoreError::SerializationError(format!(
                 "Value '{}' not found in enum '{}'",
-                s, dbt.name
+                s,
```
[the remark this answers is not in the document]
Comment (dpetrick) [reply query-engine/core/src/response_ir/internal.rs:585]: EnumTypes are now WeakRefs too, before we just build as many strong refs as needed, which changed to be more structured.

**Review by garrensmith — APPROVED** [review]
Looks good thanks.
